### PR TITLE
docs: Create documentation for update strategy

### LINKED ISSUE #207: Document how to perform an update of a control plane with sail-operator

### PR DESCRIPTION
**Pull request by fjglira**
Fixes #207

This PR adds the documentation of the two update strategies available in the sail-operator. Also, I added a useful table of contents to be able to find the information in the read me

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -75,6 +108,281 @@ The Sail-operator does not manage Gateways. You can deploy a gateway manually ei
 
 **Note:** The `IstioOperator` / `istioctl` example is separate from the Sail-operator. Setting `spec.components` or `spec.values.gateways` on your Sail-operator `Istio` resource **will not work**.
 
+## Update Strategy
+
+The sail-operator supports two update strategies to update the istio version of the control plane: `InPlace` and `RevisionBased`. The default strategy is `InPlace`.
```
Comment (FilipB) [inline docs/README.md:113]: maybe: 'to update a version of the Istio control plane'?

File: docs/README.md [hunk 2]
```diff
@@ -75,6 +108,281 @@ The Sail-operator does not manage Gateways. You can deploy a gateway manually ei
...
+9. Confirm that the new version is used in the sidecar
+
+    ```bash
+    kubectl get pod -n bookinfo -l app=details -o yaml | yq '.items[].spec.containers[].image'
+    docker.io/istio/examples-bookinfo-details-v1:1.19.1
+    docker.io/istio/proxyv2:1.21.2
+    ```
+
+### RevisionBased
+When the `RevisionBased` strategy is used, a new Istio control plane instance is created for every change to the Istio.spec.version field. The old control plane remains in place until all workloads have been moved to the new control plane instance, then, the old control plane is deleted.
```
Comment (FilipB) [inline docs/README.md:218]: Is the old control plane removed automatically?

File: docs/README.md [hunk 3]
```diff
@@ -75,6 +108,281 @@ The Sail-operator does not manage Gateways. You can deploy a gateway manually ei
...
+    ```bash
+    kubectl apply -n bookinfo -f https://raw.githubusercontent.com/istio/istio/release-1.22/samples/bookinfo/platform/kube/bookinfo.yaml
+    ```
+
+8. Confirm proxy sidecar injection version match the control plane version
+
+    ```bash
+    kubectl get pod -n bookinfo -l app=details -o yaml | yq '.items[].spec.containers[].image'
+    docker.io/istio/examples-bookinfo-details-v1:1.19.1
+    docker.io/istio/proxyv2:1.21.0
```
Comment (FilipB) [inline docs/README.md:299]: The same as above for in-place

**Review by FilipB — CHANGES_REQUESTED** [review]

File: docs/README.md (hunk 2 above)
Comment (fjglira) [reply docs/README.md:218]: Yes, it will be removed automatically when all the pods are moved to the new one

File: docs/README.md [hunk 4]
```diff
@@ -75,6 +108,281 @@ The Sail-operator does not manage Gateways. You can deploy a gateway manually ei
...
+      version: v1.21.2
+    EOF
+    ```
+
+7. Confirm the istio resource version was updated
+
+    ```bash
+    kubectl get istio -n istio-system
+    NAME      REVISIONS   READY   IN USE   ACTIVE REVISION   VERSION   AGE
+    default   1           1       1        Healthy           v1.21.2   12m
```
Comment (luksa) [inline docs/README.md:200]:
```suggestion
    $ kubectl get istio -n istio-system
    NAME      REVISIONS   READY   IN USE   ACTIVE REVISION   VERSION   AGE
    default   1           1       1        Healthy           v1.21.2   12m
```

Or remove the `$` from the other code blocks above.

File: docs/README.md [hunk 5]
```diff
@@ -75,6 +108,281 @@ The Sail-operator does not manage Gateways. You can deploy a gateway manually ei
...
+    ```bash
+    kubectl delete pods --all -n bookinfo
+    ```
+
+9. Confirm that the new version is used in the sidecar
+
+    ```bash
+    kubectl get pod -n bookinfo -l app=details -o yaml | yq '.items[].spec.containers[].image'
+    docker.io/istio/examples-bookinfo-details-v1:1.19.1
+    docker.io/istio/proxyv2:1.21.2
```
Comment (luksa) [inline docs/README.md:214]:
```suggestion
    $ kubectl get pod -n bookinfo -l app=details -o yaml | yq '.items[].spec.containers[].image'
    docker.io/istio/examples-bookinfo-details-v1:1.19.1
    docker.io/istio/proxyv2:1.21.2
```

This makes it easy to differentiate between the command and its output.

File: docs/README.md [hunk 6]
```diff
@@ -75,6 +108,281 @@ The Sail-operator does not manage Gateways. You can deploy a gateway manually ei
...
+
+Steps:
+1. Create the `istio-system` and the `istio-cni` namespaces.
+
+    ```bash
+    kubectl create namespace istio-system
+    kubectl create namespace istio-cni
+    ```
+
+2. Create the `Istio-cni` resource
```
Comment (luksa) [inline docs/README.md:131]:
```suggestion
2. Create the `IstioCni` resource
```

Comment (luksa) [reply docs/README.md:131]: But, I think we should omit `IstioCNI` here altogether. IstioCNI is not needed and mentioning it here only makes the important (upgrade) stuff less visible.

File: docs/README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (luksa) [reply docs/README.md:214]: Or just check the annotation (I think it's `istio.io/rev` or something similar).

File: docs/README.md [hunk 7]
```diff
@@ -75,6 +108,281 @@ The Sail-operator does not manage Gateways. You can deploy a gateway manually ei
...
+    ```bash
+    kubectl create namespace bookinfo
+    kubectl label namespace bookinfo istio.io/rev=default
+    kubectl apply -n bookinfo -f https://raw.githubusercontent.com/istio/istio/release-1.22/samples/bookinfo/platform/kube/bookinfo.yaml
+    ```
+
+6. Perform the update of the control plane by changing the version in the Istio resource
+
+    ```bash
+    cat <<EOF | kubectl apply -f-
```
[the remark this answers is not in the document]
Comment (luksa) [reply docs/README.md:181]: +1

Use patch to update only the `spec.version` field.

File: docs/README.md [hunk 8]
```diff
@@ -75,6 +108,281 @@ The Sail-operator does not manage Gateways. You can deploy a gateway manually ei
...
+
+15. Confirm the new version is used in the sidecars
+
+    ```bash
+    kubectl get pod -n bookinfo -l app=details -o yaml | yq '.items[].spec.containers[].image'
+    docker.io/istio/examples-bookinfo-details-v1:1.19.1
+    docker.io/istio/proxyv2:1.21.2
+    ```
+
+16. Confirm the old control plane deletion
```
Comment (luksa) [inline docs/README.md:370]: Add note saying that the old revision (and control plane) are deleted after the grace period specified in `Istio.spec.updateStrategy.inactiveRevisionDeletionGracePeriodSeconds` runs out.

**Comment by luksa** [comment]
Please check for consistency on using `$` in front of commands. I like the `$`, because it makes it easy to differentiate the command from its output. However, I hate it when there are multiple commands that I need to copy-paste.

Here, you use it sometimes and sometimes not.

**Comment by luksa** [comment]
TIL: if you use start the code block with ```console, the output gets a different color than the command:

```console
$ some-command
first line of output
second line of output
```

File: docs/README.md (hunk 8 above)
Comment (sridhargaddam) [reply docs/README.md:370]: @luksa I remember seeing that after we update the label on bookinfo namespace to point to new revision and its restarted, the old Istio revision (and control plane) pods are immediately deleted. i.e., it does not wait for `inactiveRevisionDeletionGracePeriodSeconds` 

Do you think its a bug?

File: docs/README.md [hunk 9]
```diff
@@ -75,6 +108,281 @@ The Sail-operator does not manage Gateways. You can deploy a gateway manually ei
...
+    $ kubectl get istio -n istio-system
+      NAME      REVISIONS   READY   IN USE   ACTIVE REVISION   VERSION   AGE
+      default   2           2       1        Healthy           v1.21.2   23m
+    $ kubectl get istiorevision -n istio-system
+      NAME              READY   STATUS    IN USE   VERSION   AGE
+      default-v1-21-0   True    Healthy   True     v1.21.0   27m
+      default-v1-21-2   True    Healthy   False    v1.21.2   4m45s
+    ```
+
+11. Confirm there is two control plane pods running, one for each revision
```
Comment (sridhargaddam) [inline docs/README.md:331]:
```suggestion
11. Confirm there are two control plane pods running, one for each revision
```

Comment (luksa) [reply docs/README.md:331]: @sridhargaddam We do have tests that check the grace period. Are you sure they're getting deleted immediately and not after 30s (the default grace period)?

Comment (sridhargaddam) [reply docs/README.md:331]: @luksa I was following the steps mentioned in this PR and I'm fairly sure of seeing this behavior. 

However, I believe the reason is that when the initial Istio object was created at Line 248, it was given the name default. When it is updated at Line 302, we are merely updating the same Istio object. 

Shouldn't we use a different name for `Istio` objects when using revisions?

Comment (luksa) [reply docs/README.md:331]: No, using the same object is correct. The operator takes care of creating two `IstioRevision` objects underneath.

File: docs/README.md (hunk 6 above)
Comment (fjglira) [reply docs/README.md:131]: I deleted the section, we are going to skip mentioning the istioCNI resource

File: docs/README.md [hunk 10]
```diff
@@ -75,6 +108,281 @@ The Sail-operator does not manage Gateways. You can deploy a gateway manually ei
...
+    $ kubectl get istio -n istio-system
+    NAME      READY   STATUS    IN USE   VERSION   AGE
+    default   True    Healthy   True     v1.21.0   2m
+    ```
+
+5. Create namespace `bookinfo` and deploy bookinfo application
+
+    ```bash
+    kubectl create namespace bookinfo
+    kubectl label namespace bookinfo istio.io/rev=default
```
[the remark this answers is not in the document]
Comment (fjglira) [reply docs/README.md:174]: For `InPlace` strategy we can use both labels or `istio-io/rev` or `istio-injection`. For `InPlace` make sense to use only `istio-injection=enabled`

[the remark this answers is not in the document]
Comment (luksa) [reply docs/README.md:174]: Be careful. Even if you use `InPlace`, you have to use `istio.io/rev` if the name of your `Istio` CR is anything other than `default`.

File: docs/README.md (hunk 9 above)
Comment (sridhargaddam) [reply docs/README.md:331]: There was some confusion on my end. I misread it as 30 mins and not seconds. We are good. Thanks guys!

File: docs/README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (fjglira) [reply docs/README.md:214]: Using istioctl is the best option because there is no label with the istio version used

File: docs/README.md [hunk 11]
```diff
@@ -75,6 +108,281 @@ The Sail-operator does not manage Gateways. You can deploy a gateway manually ei
...
+      docker.io/istio/proxyv2:1.21.0
+    ```
+
+13. Change the label of the `bookinfo` namespace
+
+    ```bash
+    kubectl label namespace bookinfo istio.io/rev=default-v1-21-2 --overwrite
+    ```
+
+The control plane used by the pods will be updated to the new version only after all workloads are moved to the new revision, and this will happen after the pods are restarted.
```
[the remark this answers is not in the document]
Comment (fjglira) [reply docs/README.md:354]: Done

**Comment by fjglira** [comment]
I updated all the documentation based in the comments

File: docs/README.md [hunk 12]
```diff
@@ -75,6 +108,229 @@ The Sail-operator does not manage Gateways. You can deploy a gateway manually ei
...
+    ```bash
+    cat <<EOF | kubectl apply -f-
+    apiVersion: operator.istio.io/v1alpha1
+    kind: Istio
+    metadata:
+      name: default
+    spec:
+      namespace: istio-system
+      updateStrategy:
+        type: InPlace
```
Comment (sridhargaddam) [inline docs/README.md:222]:
```suggestion
        type: RevisionBased
```

**Review by sridhargaddam — CHANGES_REQUESTED** [review]

**Comment by sridhargaddam** [comment]
@fjglira I hope you wanted to incorporate the suggestion from @luksa to use "console" instead of "bash"
https://github.com/istio-ecosystem/sail-operator/pull/224#issuecomment-2242906491

File: docs/README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (luksa) [reply docs/README.md]: But do you really need to see the version? Or do you actually want to see the *revision* that the pod is attached to?

```
kubectl get pod -n bookinfo -l app=details -oyaml | yq '.items[].metadata.annotations["istio.io/rev"]'
```

File: docs/README.md [hunk 13]
```diff
@@ -75,6 +108,229 @@ The Sail-operator does not manage Gateways. You can deploy a gateway manually ei
...
+The sail-operator supports two update strategies to update the version of the Istio control plane: `InPlace` and `RevisionBased`. The default strategy is `InPlace`.
+
+### InPlace
+When the `InPlace` strategy is used, the existing Istio control plane is updated in-place. The workloads therefore don't need to be moved from one control plane instance to another.
+
+#### Example using the InPlace strategy
+
+Prerequisites:
+* sail-operator needs to be installed.
+* Deploy `IstioCni` resource.
```
Comment (luksa) [inline docs/README.md:122]: IstioCNI is not a prerequisite. Since these aren't OpenShift-specific docs, we can skip IstioCNI completely.

File: docs/README.md [hunk 14]
```diff
@@ -75,6 +108,229 @@ The Sail-operator does not manage Gateways. You can deploy a gateway manually ei
...
+    ```
+
+4. Create namespace `bookinfo` and deploy bookinfo application
+
+    ```bash
+    kubectl create namespace bookinfo
+    kubectl label namespace bookinfo istio-injection=enabled
+    kubectl apply -n bookinfo -f https://raw.githubusercontent.com/istio/istio/release-1.22/samples/bookinfo/platform/kube/bookinfo.yaml
+    ```
+Note: if the `Istio` resource name is other than default, you need to label the namespace with the revision name.
```
Comment (luksa) [inline docs/README.md:164]:
```suggestion
Note: if the `Istio` resource name is other than `default`, you need to set the `istio.io/rev` label to the name of the 
`Istio` resource instead of adding the `istio-injection=enabled` label.
```

File: docs/README.md [hunk 15]
```diff
@@ -75,6 +108,229 @@ The Sail-operator does not manage Gateways. You can deploy a gateway manually ei
...
+
+5. Perform the update of the control plane by changing the version in the Istio resource
+
+    ```bash
+    kubectl patch istio default -n istio-system --type='merge' -p '{"spec":{"version":"v1.21.2"}}'
+    ```
+
+6. Confirm the istio resource version was updated
```
Comment (luksa) [inline docs/README.md:172]:
```suggestion
6. Confirm the `Istio` resource version was updated
```

File: docs/README.md [hunk 16]
```diff
@@ -75,6 +108,229 @@ The Sail-operator does not manage Gateways. You can deploy a gateway manually ei
...
+
+    ```bash
+    $ kubectl get istio -n istio-system
+    NAME      REVISIONS   READY   IN USE   ACTIVE REVISION   VERSION   AGE
+    default   1           1       1        Healthy           v1.21.2   12m
+    ```
+
+7. Delete `bookinfo` pod to trigger sidecar injection with the new version
```
Comment (luksa) [inline docs/README.md:180]:
```suggestion
7. Delete `bookinfo` pods to trigger sidecar injection with the new version
```

File: docs/README.md [hunk 17]
```diff
@@ -75,6 +108,229 @@ The Sail-operator does not manage Gateways. You can deploy a gateway manually ei
...
+    cat <<EOF | kubectl apply -f-
+    apiVersion: operator.istio.io/v1alpha1
+    kind: Istio
+    metadata:
+      name: default
+    spec:
+      namespace: istio-system
+      updateStrategy:
+        type: InPlace
+        inactiveRevisionDeletionGracePeriodSeconds: 30
```
Comment (luksa) [inline docs/README.md:144]: I suggest we skip this line to make the example simpler.

File: docs/README.md [hunk 18]
```diff
@@ -75,6 +108,229 @@ The Sail-operator does not manage Gateways. You can deploy a gateway manually ei
...
```
Comment (luksa) [inline docs/README.md:223]: Skip this line.

File: docs/README.md [hunk 19]
```diff
@@ -75,6 +108,229 @@ The Sail-operator does not manage Gateways. You can deploy a gateway manually ei
...
+
+3. Confirm the control plane is installed and is using the desired version.
+
+    ```bash
+    $ kubectl get istio -n istio-system
+    NAME      READY   STATUS    IN USE   VERSION   AGE
+    default   True    Healthy   True     v1.21.0   2m
+    ```
+
+4. Get the `istiorevision` name
```
Comment (luksa) [inline docs/README.md:236]:
```suggestion
4. Get the `IstioRevision` name
```

Please also check capitalization everywhere else.

File: docs/README.md [hunk 20]
```diff
@@ -75,6 +108,229 @@ The Sail-operator does not manage Gateways. You can deploy a gateway manually ei
...
+
+    ```bash
+    $ kubectl get istiorevision -n istio-system
+    NAME              READY   STATUS    IN USE   VERSION   AGE
+    default-v1-21-0   True    Healthy   False    v1.21.0   114s
+    ```
```
Comment (luksa) [inline docs/README.md:242]: Maybe add note saying that the revision name is composed of the `Istio` resource name and the version.

File: docs/README.md [hunk 21]
```diff
@@ -75,6 +108,229 @@ The Sail-operator does not manage Gateways. You can deploy a gateway manually ei
...
+    istioctl proxy-status 
+    ```
+The column `VERSION` should still match the old control plane version.
+
+12. Change the label of the `bookinfo` namespace to use the new revision
+
+    ```bash
+    kubectl label namespace bookinfo istio.io/rev=default-v1-21-2 --overwrite
+    ```
+The injected version by the control plane will not change immediately. Only will be changed after the pods are restarted, this will trigger the sidecar injection with the new version.
```
Comment (luksa) [inline docs/README.md:303]:
```suggestion
The injected version by the control plane will not change immediately. It will only be changed after the pods are restarted, this will trigger the sidecar injection with the new version.
```

File: docs/README.md [hunk 22]
```diff
@@ -75,6 +108,229 @@ The Sail-operator does not manage Gateways. You can deploy a gateway manually ei
...
+
+14. Confirm the new version is used in the sidecars
+
+    ```bash
+    istioctl proxy-status 
+    ```
+The column `VERSION` should match the updated control plane version.
+
+15. Confirm the old control plane deletion
+The old control plane will be deleted after the grace period specified in the `Istio` resource `Istio.spec.updateStrategy.inactiveRevisionDeletionGracePeriodSeconds`.
```
Comment (luksa) [inline docs/README.md:319]:
```suggestion
15. Confirm the old control plane deletion

The old control plane will be deleted after the grace period specified in the `Istio` resource `Istio.spec.updateStrategy.inactiveRevisionDeletionGracePeriodSeconds`.
```

File: docs/README.md (hunk 18 above)
Comment (luksa) [reply docs/README.md:223]: Actually, no. You mention this grace period later, so let's keep it here. But maybe still remove it from the InPlace example.

File: docs/README.md [hunk 23]
```diff
@@ -75,6 +108,229 @@ The Sail-operator does not manage Gateways. You can deploy a gateway manually ei
...
+    ```bash
+    $ kubectl get pods -n istio-system
+    NAME                                      READY   STATUS    RESTARTS   AGE
+    istiod-default-v1-21-2-7c4f4674c5-4g7n7   1/1     Running   0          94m
+    ```
+
+16. Confirm the old revision deletion
+The old revision will be deleted after the grace period specified in the `Istio` resource `Istio.spec.updateStrategy.inactiveRevisionDeletionGracePeriodSeconds`.
```
Comment (luksa) [inline docs/README.md:327]:
```suggestion
16. Confirm the old revision deletion

The old revision will be deleted after the grace period specified in the `Istio` resource `Istio.spec.updateStrategy.inactiveRevisionDeletionGracePeriodSeconds`.
```

File: docs/README.md [hunk 24]
```diff
@@ -75,6 +108,229 @@ The Sail-operator does not manage Gateways. You can deploy a gateway manually ei
...
+14. Confirm the new version is used in the sidecars
+
+    ```bash
+    istioctl proxy-status 
+    ```
+The column `VERSION` should match the updated control plane version.
+
+15. Confirm the old control plane deletion
+
+The old control plane will be deleted after the grace period specified in the `Istio` resource `Istio.spec.updateStrategy.inactiveRevisionDeletionGracePeriodSeconds`.
```
Comment (sridhargaddam) [inline docs/README.md:318]: nit: Since we are kind of repeating similar message even in Line 327, we can probably club number 15 and 16 together and show both the outputs to avoid the duplication.

**Review by sridhargaddam — APPROVED** [review]

File: docs/README.md [hunk 25]
```diff
@@ -75,6 +108,229 @@ The Sail-operator does not manage Gateways. You can deploy a gateway manually ei
 
 **Note:** The `IstioOperator` / `istioctl` example is separate from the Sail-operator. Setting `spec.components` or `spec.values.gateways` on your Sail-operator `Istio` resource **will not work**.
 
+## Update Strategy
+
+The sail-operator supports two update strategies to update the version of the Istio control plane: `InPlace` and `RevisionBased`. The default strategy is `InPlace`.
+
+### InPlace
+When the `InPlace` strategy is used, the existing Istio control plane is updated in-place. The workloads therefore don't need to be moved from one control plane instance to another.
```
Comment (luksa) [inline docs/README.md:116]:
```suggestion
When the `InPlace` strategy is used, the existing Istio control plane is replaced with a new version. The workload sidecars immediately connect to the new control plane. The workloads therefore don't need to be moved from one control plane instance to another.
```

File: docs/README.md [hunk 26]
```diff
@@ -75,6 +108,229 @@ The Sail-operator does not manage Gateways. You can deploy a gateway manually ei
...
+
+#### Example using the InPlace strategy
+
+Prerequisites:
+* sail-operator needs to be installed.
+* `istioctl` is installed.
```
[the remark this answers is not in the document]
Comment (luksa) [reply docs/README.md:122]: Also, why "sail-operator" instead of "Sail Operator"?

**Comment by luksa** [comment]
Some steps end with a period, while others don't. We should probably use periods throughout (not just for consistency, but also for accessibility - screen readers probably don't read the text properly if there's no period; especially, if there's another sentence in the next line).

File: docs/README.md [hunk 27]
```diff
@@ -75,6 +108,229 @@ The Sail-operator does not manage Gateways. You can deploy a gateway manually ei
...
+
+8. Confirm that the new version is used in the sidecar
+
+    ```bash
+    istioctl proxy-status 
+    ```
+The column `VERSION` should match the new control plane version.
+
+### RevisionBased
+When the `RevisionBased` strategy is used, a new Istio control plane instance is created for every change to the `Istio.spec.version` field. The old control plane remains in place until all workloads have been moved to the new control plane instance, then, the old control plane is deleted.
```
Comment (luksa) [inline docs/README.md:192]: I'd rephrase the second sentence to make it clear that the user is the one that has to move the workloads to the new control plane instance. Currently, some readers could misunderstand the sentence to mean that all this happens automatically.

File: docs/README.md [hunk 28]
```diff
@@ -75,6 +108,229 @@ The Sail-operator does not manage Gateways. You can deploy a gateway manually ei
...
+9. Verify the `Istio` and `IstioRevision` resources. There will be a new revision created with the new version.
+
+    ```console
+    $ kubectl get istio -n istio-system
+      NAME      REVISIONS   READY   IN USE   ACTIVE REVISION   VERSION   AGE
+      default   2           2       1        Healthy           v1.21.2   23m
+    $ kubectl get istiorevision -n istio-system
+      NAME              READY   STATUS    IN USE   VERSION   AGE
+      default-v1-21-0   True    Healthy   True     v1.21.0   27m
+      default-v1-21-2   True    Healthy   False    v1.21.2   4m45s
```
Comment (luksa) [inline docs/README.md:277]: Indentation:

```suggestion
    $ kubectl get istio -n istio-system
    NAME      REVISIONS   READY   IN USE   ACTIVE REVISION   VERSION   AGE
    default   2           2       1        Healthy           v1.21.2   23m

    $ kubectl get istiorevision -n istio-system
    NAME              READY   STATUS    IN USE   VERSION   AGE
    default-v1-21-0   True    Healthy   True     v1.21.0   27m
    default-v1-21-2   True    Healthy   False    v1.21.2   4m45s
```

File: docs/README.md [hunk 29]
```diff
@@ -75,6 +108,229 @@ The Sail-operator does not manage Gateways. You can deploy a gateway manually ei
...
+    ```
+
+10. Confirm there are two control plane pods running, one for each revision
+
+    ```console
+    $ kubectl get pods -n istio-system
+      NAME                                      READY   STATUS    RESTARTS   AGE
+      istiod-default-v1-21-0-69d6df7f9c-grm24   1/1     Running   0          28m
+      istiod-default-v1-21-2-7c4f4674c5-4g7n7   1/1     Running   0          6m9s
```
Comment (luksa) [inline docs/README.md:286]:
```suggestion
    $ kubectl get pods -n istio-system
    NAME                                      READY   STATUS    RESTARTS   AGE
    istiod-default-v1-21-0-69d6df7f9c-grm24   1/1     Running   0          28m
    istiod-default-v1-21-2-7c4f4674c5-4g7n7   1/1     Running   0          6m9s
```

File: docs/README.md [hunk 30]
```diff
@@ -75,6 +108,229 @@ The Sail-operator does not manage Gateways. You can deploy a gateway manually ei
...
+    istioctl proxy-status 
+    ```
+The column `VERSION` should still match the old control plane version.
+
+12. Change the label of the `bookinfo` namespace to use the new revision
+
+    ```bash
+    kubectl label namespace bookinfo istio.io/rev=default-v1-21-2 --overwrite
+    ```
+The injected version by the control plane will not change immediately. It will only be changed after the pods are restarted, this will trigger the sidecar injection with the new version.
```
Comment (luksa) [inline docs/README.md:301]:
```suggestion
The existing workload sidecars will continue to run and will remain connected to the old control plane instance. The sidecars will not be replaced with a new version until the pods are deleted and recreated.
```

File: docs/README.md [hunk 31]
```diff
@@ -75,6 +108,229 @@ The Sail-operator does not manage Gateways. You can deploy a gateway manually ei
...
+The old control plane will be deleted after the grace period specified in the `Istio` resource `Istio.spec.updateStrategy.inactiveRevisionDeletionGracePeriodSeconds`.
+    ```console
+    $ kubectl get pods -n istio-system
+    NAME                                      READY   STATUS    RESTARTS   AGE
+    istiod-default-v1-21-2-7c4f4674c5-4g7n7   1/1     Running   0          94m
+    ```
+
+16. Confirm the old revision deletion
+
+The old revision will be deleted after the grace period specified in the `Istio` resource `Istio.spec.updateStrategy.inactiveRevisionDeletionGracePeriodSeconds`.
```
Comment (luksa) [inline docs/README.md:327]: The indentation doesn't look alright here and in some of the previous bullets.

Comment (luksa) [reply docs/README.md:327]: Also, the code block seems to be broken here and in step 15.

File: docs/README.md (hunk 26 above)
[the remark this answers is not in the document]
Comment (fjglira) [reply docs/README.md:122]: In all the document we are naming the operator as `Sail-operator`

[the remark this answers is not in the document]
Comment (luksa) [reply docs/README.md]: Why are we doing this? We're not calling it "Sail-operator", but "sail-operator". But both forms look off. They look like a binary name or deployment/pod name. But that's an implementation detail. We're talking about the operator, not binary/deployment/pod. And the operator's name is "Sail Operator". That's also how it's listed in the OperatorHub catalog.
